test/parallel/test-debugger-util-regression.js

```
@@ -0,0 +1,65 @@
+'use strict';
+const path = require('path');
+const spawn = require('child_process').spawn;
+const assert = require('assert');
+
+const common = require('../common');
+
+const fixture = path.join(
+  common.fixturesDir,
+  'debugger-util-regression-fixture.js'
+);
+
+const args = [
+  'debug',
+  fixture
+];
+
+const proc = spawn(process.execPath, args, { stdio: 'pipe' });
+
+function cleanup() {
+  process.exitCode = 2;
+  proc.kill();
+}
+
+const timer = setTimeout(cleanup, common.platformTimeout(2000));
+
+let stdout = '';
+let stderr = '';
+
+let nextCount = 0;
+
+proc.stdout.on('data', (data) => {
+  stdout += data;
+  if (stdout.includes('> 1') && nextCount < 1 ||
+      stdout.includes('> 2') && nextCount < 2 ||
+      stdout.includes('> 3') && nextCount < 3) {
+    nextCount++;
+    proc.stdin.write('n\n');
+  }
+  else if (stdout.includes('{ a: \'b\' }')) {
+    clearTimeout(timer);
+    proc.stdin.write('.exit\n');
+  }
+  else if (stdout.includes('program terminated')) {
+    // Catch edge case present in v4.x
+    process.exitCode = 1;
+    clearTimeout(timer);
+    proc.kill();
+  }
+});
+
+proc.stderr.on('data', (data) => stderr += data);
+
+proc.stdin.on('error', (err) => {
```

Why don't you let the error bubble up?  If there is a good reason, can you add a comment explaining it?